core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -415,21 +415,17 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime) extends Logg
 
   // Copy the subset of properties that are relevant to Logs
   // I'm listing out individual properties here since the names are slightly different in each Config class...
-  private def copyKafkaConfigToLog(serverProps: java.util.Map[String, Object]): java.util.Map[String, Object] = {
+  private def copyKafkaConfigToLog(kafkaConfig: KafkaConfig): java.util.Map[String, Object] = {
```

Comment:
... although that would be a very brittle test as well - since it depends on exercising a config parameter (hours/minutes) that derives the final `LogConfig` retention parameter (millis). So we could just as well just test this copy utility. That would still be flaky since we could add more derived configs over time. i.e., we basically need 100% coverage on all configs in `LogConfig`